src/dune_rules/merlin/merlin.ml

```diff
@@ -394,7 +439,7 @@ module Unprocessed = struct
       Path.Set.singleton @@ obj_dir_of_lib `Private mode (Obj_dir.of_local obj_dir)
     in
     let flags = Ocaml_flags.get flags mode in
-    let extensions = Dialect.DB.extensions_for_merlin dialects in
+    let for_merlin = Dialect.DB.for_merlin dialects in
```

Comment:
Nit: could use a record pattern here `let {Dialect.DB.extensions; for_merlin} = ...` and use punning below.